main/jlp/net/stream_client.cpp

```
@@ -0,0 +1,403 @@
+
+  while (st->running.load()) {
+    int sock = connect_to_server(*st);
+    if (sock < 0) {
+      for (int waited = 0; waited < backoff_ms && st->running.load(); waited += 100) {
+        vTaskDelay(pdMS_TO_TICKS(100));
+      }
+      backoff_ms = std::min(backoff_ms * 2, kMaxBackoffMs);
+      continue;
+    }
+
+    if (had_session) st->reconnects.fetch_add(1);
```

Clang (14.0.6)

[warning] statement should be inside braces

(readability-braces-around-statements)